clang/test/SemaCXX/warn-unused-filescoped.cpp

```diff
@@ -76,10 +76,33 @@ struct S {
   struct SVS : public VS {
     void vm() { }
   };
+
+  struct CS {
```

Comment:
When switching the diagnostic approach, be sure to add test coverage for unused copy/move constructors, copy/move assignment, and destructors (you can get an unused destructor if you only use `new` to allocate the class rather than stack allocate it).